src/partisan_hyparview_peer_service_manager.erl

```diff
@@ -531,23 +569,29 @@ handle_message({neighbor_accepted, Peer, Tag, Sender},
             %% No more peers are suspected; no change.
             State0;
         Random ->
-            %% Remove from suspected.
-            Suspected = remove_from_suspected(Random, Suspected0),
+            case is_addable(DisconnectId, Peer, SentMessageMap) of
+                true ->
+                    %% Remove from suspected.
+                    Suspected = remove_from_suspected(Random, Suspected0),
 
-            %% Remove from reserved.
-            Reserved = remove_from_reserved(Random, Reserved0),
+                    %% Remove from reserved.
+                    Reserved = remove_from_reserved(Random, Reserved0),
 
-            %% Remove from active view.
-            Active = remove_from_active_view(Random, Active0),
+                    %% Remove from active view.
+                    Active = remove_from_active_view(Random, Active0),
 
-            %% Add to passive.
-            State1 = add_to_passive_view(
-                       Random, State0#state{suspected=Suspected,
-                                            reserved=Reserved,
-                                            active=Active}),
+                    %% Add to passive.
+                    State1 = add_to_passive_view(
+                               Random, State0#state{suspected=Suspected,
+                                                    reserved=Reserved,
+                                                    active=Active}),
 
-            %% Add to active view.
-            add_to_active_view(Peer, Tag, State1)
+                    %% Add to active view.
+                    add_to_active_view(Peer, Tag, State1);
+                false ->
+                    %% @todo Need to send the disconnect message here?
```

Comment:
Before a node sent the `neighbor_accepted` message to the peer, it added the peer to its active view. So, if the message is not valid for adding, I think we need to let the node know this and it can remove the peer from the active view.